src/NodeApi/Runtime/NativeLibrary.cs

```diff
@@ -58,8 +96,28 @@ public static nint Load(string libraryName)
     /// <returns>The address of the symbol.</returns>
     public static nint GetExport(nint handle, string name)
     {
+        if (handle == 0)
+            throw new ArgumentNullException(nameof(handle));
+        if (name is null)
+            throw new ArgumentNullException(nameof(name));
+
 #if NETFRAMEWORK || NETSTANDARD
-        return GetProcAddress(handle, name);
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return GetProcAddress(handle, name);
```

Comment:
I wonder if we should also add the error handing for Windows platform here too.
If you cannot do it, then please add a TODO here.

Reply:
I implemented the error handling. However, I am not adding messages to the exceptions. I think I'll add those later. But they're of limited use as far as I can see: this code path is only used in the new fallback resolver for Framework, which is preempted by a File.Exists check. And we also would be shipping those libraries. So, pretty limited usage. But I'll add em anyways.

Reply:
Messages added.